pkg/api/core/cloudprovider/aws/identitymanager.go

```diff
@@ -62,7 +62,7 @@ func (s *identityManagerCloudProvider) CreateIdentityPoolClient(opts api.CreateI
 		return nil, fmt.Errorf("retrieving identity pool client outputs: %w", err)
 	}
 
-	secret, err := cognito.New(s.provider).UserPoolClientSecret(c.ClientID, opts.UserPoolID)
+	secret, err := cognito.New(s.provider, nil).UserPoolClientSecret(c.ClientID, opts.UserPoolID)
```

Comment:
Sending nil is somewhat dangerous, as it may cause future panics, and we put the more of the responsibility to avoid this on the developer.

One way to avoid this can be to simply instantiate two cognito instances, one with eu-west provider, and one with us-east. However, if we want to emphasize that we only want one operation (`DeleteAuthCert`) on the us-east cognito, we can make two different structs, `EuWestCognito` and a `UsEastCognito`. The `UsEastCognito` only needs to contain the `DeleteAuthCert` method.

And maybe `UsCognito` should be named more after the functionality it provides, as @deifyed suggests. `CertDeleterCognito` or something.

But maybe this is overkill, thoughts?

Reply:
Hmm, went this route to avoid too much complexity, but i see your point - I'll try to make a new version of cognito instead :)

Reply:
We can ignore this for now just to get it out - but then we have to make sure not to forget it when rewriting it the future. If we make a task on it with this as a checkbox/todo, it should be good enough.